lib/mayaUsd/render/vp2RenderDelegate/points.cpp

```diff
@@ -0,0 +1,1071 @@
+
+    const HdRenderIndex& renderIndex = sceneDelegate->GetRenderIndex();
+
+    const auto& primvarSourceMap = _pointsSharedData._primvarSourceMap;
+
+    const MHWRender::MGeometry::DrawMode drawMode = renderItem->drawMode();
+
+    // The bounding box item uses a globally-shared geometry data therefore it
+    // doesn't need to extract index data from topology. Points use non-indexed
+    // draw.
+    const bool isBoundingBoxItem = (drawMode == MHWRender::MGeometry::kBoundingBox);
+
+#ifdef MAYA_NEW_POINT_SNAPPING_SUPPORT
+    constexpr bool isPointSnappingItem = false;
+#else
+    constexpr bool isPointSnappingItem = true;
+#endif
+
+    if (desc.geomStyle == HdPointsGeomStylePoints) {
+        // Prepare normals buffer.
+        if (itemDirtyBits & (HdChangeTracker::DirtyNormals | HdChangeTracker::DirtyDisplayStyle)) {
+            VtVec3fArray normals;
+
+            const auto it = primvarSourceMap.find(HdTokens->normals);
+            if (it != primvarSourceMap.end()) {
+                const VtValue& value = it->second.data;
+                if (ARCH_LIKELY(value.IsHolding<VtVec3fArray>())) {
+                    normals = value.UncheckedGet<VtVec3fArray>();
+                }
+            }
+
+            // Using a zero vector to indicate requirement of camera-facing
+            // normals when there is no authored normals.
+            if (normals.empty()) {
+                normals.push_back(GfVec3f(0.0f, 0.0f, 0.0f));
+            }
+
+            normals = _BuildInterpolatedArray(_pointsSharedData._points.size(), normals);
+
+            if (!_pointsSharedData._normalsBuffer) {
+                const MHWRender::MVertexBufferDescriptor vbDesc(
+                    "", MHWRender::MGeometry::kNormal, MHWRender::MGeometry::kFloat, 3);
+
+                _pointsSharedData._normalsBuffer.reset(new MHWRender::MVertexBuffer(vbDesc));
+            }
+
+            unsigned int numNormals = normals.size();
+            if (_pointsSharedData._normalsBuffer && numNormals > 0) {
+                void* bufferData = _pointsSharedData._normalsBuffer->acquire(numNormals, true);
+                if (bufferData) {
+                    memcpy(bufferData, normals.cdata(), numNormals * sizeof(GfVec3f));
+                    _CommitMVertexBuffer(_pointsSharedData._normalsBuffer.get(), bufferData);
+                }
+            }
+        }
+
+        // Prepare widths buffer.
+        if (itemDirtyBits & (HdChangeTracker::DirtyPrimvar | HdChangeTracker::DirtyDisplayStyle)) {
+            VtFloatArray widths;
+
+            const auto it = primvarSourceMap.find(HdTokens->widths);
+            if (it != primvarSourceMap.end()) {
+                const VtValue& value = it->second.data;
+                if (value.IsHolding<VtFloatArray>()) {
+                    widths = value.UncheckedGet<VtFloatArray>();
+                }
+            }
+
+            if (widths.empty()) {
+                widths.push_back(1.0f);
```

Comment:
Does Storm have the same default value? Maybe we can name it in a constexpr to make it obvious?